### PR TITLE
Support running the linter on JitBuilder

### PR DESCRIPTION
**Pull request by mgaudet**
This commit adds the makefile required to run the linter on JitBuilder.

Signed-off-by: Matthew Gaudet <magaudet@ca.ibm.com>

### CODE REVIEW

File: jitbuilder/linter.mk [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+#
+# PLATFORM=foo-bar-clang make -f linter.mk 
+#
+
+.PHONY: linter
+linter::
+
+ifeq ($(PLATFORM),ppc64-linux64-clangLinter)
+    export LLVM_CONFIG?=/tr/llvm_checker/ppc-64/sles11/bin/llvm-config
+    export CC_PATH?=/tr/llvm_checker/ppc-64/sles11/bin/clang
```
Comment (charliegracie) [inline jitbuilder/linter.mk]: Are these specific paths required?

Comment (mgaudet) [reply jitbuilder/linter.mk]: Likely no :) Good catch, I'll drop them.

**Comment by 0xdaryl** [comment]
The logic and extensible class corrections herein seem reasonable to me.

**Comment by charliegracie** [comment]
Since @0xdaryl says the compiler component changes are ok I will merge as I think the rest of the changes are ready to go.

**Pull request merged**